### PR TITLE
Unit test scripts cleanup

### PR DESCRIPTION
**Pull request by jfboismenu**
Cleans up our unit test scripts. It adds headers where they were missing, updates them where the copyright was expired, removes -3 and removes scripts that are not required.

### CODE REVIEW

File: tests/run_auth_tests.sh [hunk 1]
```diff
@@ -1,18 +0,0 @@
-#!/bin/bash
```
Comment (jfboismenu) [inline tests/run_auth_tests.sh]: Allows to run a subset of the tests, related to authentication. This list is not up to date, so getting rid of it.

File: tests/run_integration_tests.bat [hunk 2]
```diff
@@ -1,3 +0,0 @@
-@echo off
```
Comment (jfboismenu) [inline tests/run_integration_tests.bat]: Integration tests were meant to test the return code of the tank command, but this is increasingly not relevant with the phasing out of the tank command when using the webapp.

File: tests/README.md [hunk 3]
```diff
@@ -34,6 +34,12 @@ To run tests without using run_tests script:
 
     `$ unit2 discover core/tests`
 
+
+Compile sources with Python 3
+-----------------------------
+To make sure the tank source code can be compiled with Python 3, run the compile_python3.* scripts. Note that the unit tests cannot
+be executed right now with Python 3.
+
```
Comment (manneohrstrom) [inline tests/README.md]: Noticed the readme further down is a bit weird - maybe this is a good time to adjust?

image

Comment (manneohrstrom) [reply tests/README.md]: We also recommend importing `*` which i think we probably want to move away from:

image

Maybe you could do a quick pass just to double check the readme looks up to date with what we typically do today test wise?

File: tests/run_integration_tests.py [hunk 4]
```diff
@@ -1,112 +0,0 @@
...
-    _test_return_value(expected_result, args)
-    _test_return_value(expected_result, args + " shell")
-
-
-def command_line_authentication_tests():
-    """
-    Runs command line authentication tests.
-    """
-    # missing param
-    test_return_value(9, "--script-name=test")
```
Comment (manneohrstrom) [inline tests/run_integration_tests.py]: do we need to add these somewhere else? Ok to just remove?

File: tests/run_tests.sh [hunk 5]
```diff
@@ -9,5 +9,10 @@
...
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
+#
+# Allows to run the unit tests against the python executable. The actual version is determined
+# by the user's environment.
+#
+
 find . -name "*.pyc" -delete
-python -3 `dirname $0`/run_tests.py $*
+python `dirname $0`/run_tests.py $*
```
Comment (manneohrstrom) [inline tests/run_tests.sh]: shoudn't we add a `-3` argument to the travis script now if re move it here? So we still get those warnings reported by CI?

Comment (manneohrstrom) [reply tests/run_tests.sh]: i just tried it and i guess it won't matter - the travis py3 task takes care of it completely separately!

**Comment by jfboismenu** [comment]
Merged on master: https://github.com/shotgunsoftware/tk-core/commit/08a16abe1ddb990f6b8d86ef9ab48e707d3f036f

**Pull request closed without merging**